### PR TITLE
Refs 15766: update shared_mutex docs

### PR DESCRIPTION
**Pull request by MiguelBarro**
Documentation update associated with: https://github.com/eProsima/Fast-DDS/pull/2976

### CODE REVIEW

**Comment by MiguelBarro** [comment]
@richiprosima Please I have rebased my Fast-DDS branch to fix docs CI, test docs for me 😅

File: docs/installation/configuration/cmake_options.rst [hunk 1]
```diff
@@ -128,10 +128,24 @@ dependency on other options.
...
+          Fast-DDS will test the framework STL implementation (if available) and will only use it if
+          it enforces ``PTHREAD_RWLOCK_PREFER_READER_NP``. Otherwise it will automatically fallback to a custom
+          implementation.
+          This flag will enforce the use of the custom implementation in all cases.
+          Note that setting the flag ``OFF`` will not prevent the use of the custom implementation in those
+          frameworks that default to ``PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP``.
           This flag prevents thread sanitizer reports on GCC/Clang STL implementations.
         - ``ON`` ``OFF``
-        - ``OFF``
+        - ``OFF`` (linux & Mac), ``ON`` (Windows)
```
Comment (JLBuenoLopez-eProsima) [inline docs/installation/configuration/cmake_options.rst:146]: Nitpick: I suggest capitalizing Linux. In this case, maybe the addition to the spelling word list is not required (I think is the word list it case sensitive).

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM though there is a small nitpick.

File: docs/installation/configuration/cmake_options.rst (hunk 1 above)
Comment (MiguelBarro) [reply docs/installation/configuration/cmake_options.rst]: acknowledge

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
:+1:

**Comment by MiguelCompany** [comment]
@MiguelBarro This one has conflicts ....

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**